lib/Dialects/LinalgTransform/Transforms/FunctionalTransforms.cpp

```diff
@@ -0,0 +1,222 @@
+//===-- FunctionalTransforms.cpp - Functional Linalg Transforms -----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Dialects/LinalgTransform/FunctionalTransforms.h"
+#include "mlir/Dialect/Linalg/Transforms/HoistPadding.h"
+#include "mlir/Dialect/SCF/Transforms.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+namespace mlir {
+namespace linalg {
+
+//===----------------------------------------------------------------------===//
+// BEGIN copied from Linalg/Transforms/Transforms.cpp
+//===----------------------------------------------------------------------===//
+
+/// Try to peel a loop `op` and return the new result.
+// TODO: Add support for scf.parallel and affine.for loops.
+static SmallVector<Value, 4> peelLoop(RewriterBase &rewriter, Operation *op) {
+  return llvm::TypeSwitch<Operation *, SmallVector<Value, 4>>(op)
+      .Case<scf::ForOp>([&](scf::ForOp forOp) {
+        scf::ForOp partialIteration;
+        if (succeeded(scf::peelAndCanonicalizeForLoop(rewriter, forOp,
+                                                      partialIteration)))
+          return partialIteration->getResults();
+        assert(!partialIteration && "expected that loop was not peeled");
+        return forOp->getResults();
+      })
+      .Default([&](Operation *op) { return op->getResults(); });
+}
+
+/// Try to peel a TiledLoopOp and return the new result.
+static SmallVector<Value, 4> peelLoop(RewriterBase &rewriter,
+                                      TiledLoopOp tiledLoop, int64_t idx) {
+  assert(idx < static_cast<int64_t>(tiledLoop.iterator_types().size()) &&
+         "requested peeling of non-existing loop");
+  TiledLoopOp result;
+  if (succeeded(peelAndCanonicalizeTiledLoop(rewriter, tiledLoop, idx, result)))
+    return result->getResults();
+  assert(!result && "expected that loop was not peeled");
+  return tiledLoop->getResults();
+}
+
+/// Peel loops after tiling.
+static void peelLoops(RewriterBase &rewriter, TiledLinalgOp &res,
```

Comment:
Exposing this function in core.
OTOH if you just wrap the unique LinalgTilingPattern we should be in a better place.